guillotina/db/strategies/simple.py

```diff
@@ -10,29 +10,12 @@
 
 @configure.adapter(for_=ITransaction, provides=IDBTransactionStrategy, name="simple")
 class SimpleStrategy(BaseStrategy):
-    """
-    Do not attempt to resolve conflicts but detect for them
-    """
 
     async def tpc_begin(self):
         await self.retrieve_tid()
         if self._transaction._db_txn is None:
             await self._storage.start_transaction(self._transaction)
 
-    async def tpc_vote(self):
-        if not self.writable_transaction:
-            return True
-
-        current_tid = await self._storage.get_current_tid(self._transaction)
-        if current_tid > self._transaction._tid:
-            logger.warn(
-                f"Could not resolve conflicts in TID: {self._transaction._tid}\n"
-                f"Conflicted TID: {current_tid}"
-            )
-            return False
-
-        return True
```

Comment:
I reviewed the git history and I this was added before adding the same logic at the _store_object(), so I think that now this isn't needed. In fact, this check produces a failure when using nested transactions. The outer transaction always fails because the inner one changed the transaction id.

Reply:
This logic is a bit different than what is in store_object. Although, I don't really think simple is used at all... It is pretty "dumb" of an approach and will fail many concurrent write situations from what I can see.

Reply:
I was considering to start using it. The reason is that we have some endpoints that modify guillotina objects and do some sql inserts to another table. Without using the strategy simple we found the hard way two problems:

1. (Obviously) the asyncpg execute is done before the transaction, so it's inserted whether the guillotina transaction is commited or aborted
2. In case of TID Conflict the request is retried the 3 or 4 times so we end inserting the thing to other table 4 times

We mitigated the second point with some constraints and `INSERT INTO ... ON CONFLICT DO NOTHING` but in our case it would be great to be able to join the guillotina transaction with the real pg transaction

Reply:
Another option is to start the transaction early. We do this and it works seamlessly.

Something like:

```
    txn = get_current_transaction()
    stg = txn.storage
    if txn._db_txn is None:
        await stg.start_transaction(txn)
```

Performance on requests in these transaction blocks will be worse but it might be what you want.

Reply:
This is exactly what we want to do but we were thinking to add it globally.

> Performance on requests in these transaction blocks will be worse but it might be what you want.

We don't think that I'll be a big problem at our scale